internal/driver/db.go

```diff
@@ -1,142 +1,117 @@
 package driver
 
 import (
-	"database/sql"
-	"fmt"
-	"strconv"
-)
+	"sync"
 
-const (
-	qlDatabaseDriverName = "ql2"
-	qlDatabaseName       = "memory://deviceVirtual.db"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
 )
 
-var data struct {
-	DeviceName          string
+type data struct {
 	CommandName         string
-	DeviceResourceName  string
-	EnableRandomization string
+	EnableRandomization bool
 	DataType            string
 	Value               string
 }
 
+// Global: the data we are tasked with storing
+// Outer key: device name, inner key: resource name, value: resource info
+var resources map[string]map[string]data
+
+// Golang maps alone are not safe for concurrent access
+var resources_lock sync.RWMutex
+
 type db struct {
-	driverName  string
-	path        string
-	name        string
-	connection  *sql.DB
-	transaction *sql.Tx
+	driverName string
+	name       string
 }
 
 func getDb() *db {
 	return &db{
-		driverName: qlDatabaseDriverName,
-		name:       qlDatabaseName,
+		driverName: "Map",
+		name:       "Transient",
 	}
 }
 
-func (db *db) openDb() error {
-	d, err := sql.Open(db.driverName, db.name)
-	if err == nil {
-		db.connection = d
-	}
-	return err
+func (db *db) init() error {
+	resources_lock.Lock()
+	defer resources_lock.Unlock()
+	resources = make(map[string]map[string]data)
+	return nil
 }
 
-func (db *db) startTransaction() error {
-	if db.connection == nil {
-		return fmt.Errorf("Lost DB connection, forgot to openDb()? ")
-	}
-	if tx, err := db.connection.Begin(); err != nil {
-		return err
-	} else {
-		db.transaction = tx
-		return nil
-	}
+func (db *db) openDb() error {
```

Comment:
we can really remove this function and all the code calling this function

Reply:
Removed.